### PR TITLE
Split MountPoint into platform specific

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

A `TODO` I've had for a while now. This splits the MountPoints structure which now has three platform specific fields into common and platform-specific pieces.

### CODE REVIEW

**Comment by thaJeztah** [comment]
LOL, actually, it's green!

**Comment by jhowardmsft** [comment]
Works for me! Removing "Work in progress". Ready for review :smile:

File: daemon/volumes_unix.go [hunk 1]
```diff
@@ -76,3 +77,19 @@ func setBindModeIfNull(bind *volume.MountPoint) *volume.MountPoint {
 	}
 	return bind
 }
+
+// relabelIfNeeded is platform specific processing to relabel a bind.
+func relabelIfNeeded(container *container.Container, bind *volume.MountPoint) error {
+	if label.RelabelNeeded(bind.Mode) {
+		if err := label.Relabel(bind.Source, container.MountLabel, label.IsShared(bind.Mode)); err != nil {
```
Comment (thaJeztah) [inline daemon/volumes_unix.go]: Guess the `if` is a bit redundant now, and can be `return label.Relabel(.....`

File: volume/volume_windows.go [hunk 2]
```diff
@@ -197,3 +208,9 @@ func ValidMountMode(mode string) bool {
 func ReadWrite(mode string) bool {
 	return rwModes[strings.ToLower(mode)]
 }
+
+// GetCopyMode gets the copy mode from the mode string for mounts. Not supported
+// on Windows - just return false,false
```
Comment (thaJeztah) [inline volume/volume_windows.go]: Will this be supported at some point? (Just curious, as it seems like a big omission if Windows won't have it)

**Comment by thaJeztah** [comment]
ping @cpuguy83 PTAL

File: volume/volume_windows.go (hunk 2 above)
Comment (jhowardmsft) [inline volume/volume_windows.go]: Never say never, but no time soon

**Comment by thaJeztah** [comment]
Needs a rebase @jhowardmsft

**Comment by jhowardmsft** [comment]
Rebased

**Comment by jhowardmsft** [comment]
ping @cpuguy83

**Comment by LK4D4** [comment]
@jhowardmsft need rebase again :(

**Comment by cpuguy83** [comment]
Not a huge fan of splitting these up. Would rather have omitempty's.

**Comment by LK4D4** [comment]
@jhowardmsft what do you think about omitemptys?

**Comment by cpuguy83** [comment]
Or platform specific config like:

```
type MountPoint
    LinuxConfig *LinuxMountConfig
    WindowsCOnfig *WindowsMountConfig
}
```

**Comment by thaJeztah** [comment]
ping @jhowardmsft were you still working on this one, or should we close for now?

**Pull request closed without merging**